source/virtualBuffers/webKit.py

```diff
@@ -49,7 +49,17 @@ def __init__(self,rootNVDAObject):
 		super(WebKit,self).__init__(rootNVDAObject,backendName="webKit")
 
 	def __contains__(self,obj):
-		return winUser.isDescendantWindow(self.rootNVDAObject.windowHandle, obj.windowHandle)
+		if not winUser.isDescendantWindow(self.rootNVDAObject.windowHandle, obj.windowHandle):
+			return False
+		# #15653: The list items within combo boxes should not be classed as part of the browse mode document.
+		# Otherwise arrowing to them will switch back to browse mode.
+		if obj.role == controlTypes.Role.STATICTEXT:
+			parent = obj.parent
+			if parent and parent.role == controlTypes.Role.LIST:
+				parent = parent.parent
+				if parent and parent.role == controlTypes.Role.COMBOBOX:
+					return False

```

Comment:
Thoughts on this form? I think its easier to understand the structure of the controls like this.

```suggestion
			parent = obj.parent
			grandParent = parent.parent
			if (
				parent
				and parent.role == controlTypes.Role.LIST
				and grandParent
				and grandParent.role == controlTypes.Role.COMBOBOX
			):
				return False

```

Reply:
Agreed it is easier to parse, but it would be more costly, as you would be needlessly fetching grandparent in the vast majority of cases. And NvDAObject instantiation is not cheap. No doubt we could improve this, but right now that is the case.